metricbeat/module/system/raid/raid.go

```diff
@@ -67,33 +69,67 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
 		return nil, err
 	}
 
+	sysMountPoint := filepath.Join(config.MountPoint, "/sys/block")
+
 	return &MetricSet{
 		BaseMetricSet: base,
 		fs:            fs,
+		sysfs:         sysMountPoint,
 	}, nil
 }
 
+//Annoyingly, different linux subsystems report size size using different blocks
+// /proc/mdstat /proc/partitions and mdadm (Via the BLKGETSIZE64 ioctl call) count "size" as the number of 1024-byte blocks.
+// /sys/block/md*/size uses 512-byte blocks. As does /sys/block/md*/md/sync_completed
+//convert the 512-byte blocks to 1024 byte blocks to maintain how this metricset "used to" report size
+func blockto1024(b int64) int64 {
+	return b / 2
+}
+
 // Fetch fetches one event for each device
 func (m *MetricSet) Fetch(r mb.ReporterV2) {
-	stats, err := m.fs.ParseMDStat()
+	devices, err := blockinfo.ListAllMDDevices(m.sysfs)
```

Comment:
Some ideas to make this more independant of the OS and the raid subsystem:
* Make blockinfo an object that implements an interface
* This object lists directly the objects, so nothing like `GetMDDevice` is needed later

So at the end you can have something like:
```
	devices, err := m.blockinfo.ListAll()
	...
	for _, device := range devices {
	    disks := device.Disks()
	    ...
	}
```

Also take into account that even on Linux there can be different RAID implementations.